File: Boxuan/16_01_18/Find_Minimum_in_Rotated_Sorted_Array.py
class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
            
        left, right = 0, len(nums) - 1
        while left + 1 < right:
            mid = left + (right - left) / 2
            if nums[mid] >= nums[right]:
                left = mid
            else:
                right = mid
                
        return min(nums[right], nums[left])

File: Boxuan/16_01_18/Find_Minimum_in_Rotated_Sorted_Array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        if (nums.size() == 0){
            return 0;
        }
        int left = 0;
        int right = nums.size() - 1;
        
        while (left + 1 < right){
            int mid = left + (right - left) / 2;
            if (nums[right] <= nums[mid]){
                left = mid;
            }else{
                right = mid;
            }
        }
        return min(nums[right], nums[left]);
    }
};